docs/DeveloperGuide.md

```diff
@@ -0,0 +1,123 @@
+ 4. Generates a [`FileResult`](/src/main/java/reposense/authorship/model/FileResult.java) for each file, which consolidates the authorship results into a *Map* of each author's line contribution to the file.
+ 5. Uses [`FileResultAggregator`](/src/main/java/reposense/authorship/FileResultAggregator.java) to aggregate all `FileResult` into an `AuthorshipSummary`.
+
+
+### ReportGenerator
+[`ReportGenerator`](/src/main/java/reposense/report/ReportGenerator.java),
+ 1. Uses `GitDownloader` API to download the repository from *GitHub*.
+ 2. Copies the template files into the designated output directory.
+ 3. Uses `CommitReporter` and `AuthorshipReporter` to produce the commit and authorship summary respectively.
+ 4. Generates the `JSON` files needed to generate the `HTML` dashboard.
+
+
+### System
+`System` contains the classes that interact with the Operating System and external processes.
+ * [`CommandRunner`](/src/main/java/reposense/system/CommandRunner.java) creates processes that executes commands on the terminal. It consists of many *git* commands.
+ * [`LogsManager`](/src/main/java/reposense/system/LogsManager.java) uses the `java.util.logging` package for logging. The `LogsManager` class is used to manage the logging levels and logging destinations. Log messages are output through: `Console` and to a `.log` file.
+
+
+### Model
+`Model` holds the data structures that are commonly used by the different aspects of *RepoSense*.
+ * [`Author`](/src/main/java/reposense/model/Author.java) stores the `GitHub ID` of an author. Any contributions or commits made by the author, using his/her `GitHub ID` or aliases, will be attributed to the same `Author` object. It is used by `AuthorshipReporter` and `CommitsReporter` to attribute the commit and file contributions to the respective authors.
+ * [`CliArguments`](/src/main/java/reposense/model/CliArguments.java) stores the parsed command line arguments supplied by the user. It contains the configuration settings such as the CSV config file to read from, the directory to output the report to, and date range of commits to analyze. These configuration settings are passed into `RepoConfiguration`.
+ * [`RepoConfiguration`](/src/main/java/reposense/model/RepoConfiguration.java) stores the configuration information from the CSV config file for a single repository, which are the repository's orgarization, name, branch, list of authors to analyse, date range to analyze commits and files from `CliArguments`. 
+ These configuration information are used by:
+    - `GitDownloader` to determine which repository to download from and which branch to check out to.
+    - `AuthorshipReporter` and `CommitsReporter` to determine the range of commits and files to analyze.
+    - `ReportGenerator` to determine the directory to output the report.
+
+
+## HTML Dashboard
+
+ ![Dashboard](images/dashboard-architeture.png)
```

Comment:
May be more appropriate to draw this as a folder/file hierarchy? Not sure. 
Also, indicate the meaning of the arrows. `include`? `uses`?